repair/row_level.cc

```diff
@@ -596,16 +597,22 @@ class repair_writer : public enable_lw_shared_from_this<repair_writer> {
                 auto& cs = t->get_compaction_strategy();
                 const auto adjusted_estimated_partitions = cs.adjust_partition_estimate(metadata, estimated_partitions);
                 sstables::offstrategy offstrategy = is_offstrategy_supported(reason);
+                bool auto_trigger = offstrategy && (reason == streaming::stream_reason::repair);
                 auto consumer = cs.make_interposer_consumer(metadata,
-                        [t = std::move(t), &view_update_gen, use_view_update_path, adjusted_estimated_partitions, offstrategy] (flat_mutation_reader reader) {
+                        [t = std::move(t), &view_update_gen, use_view_update_path, adjusted_estimated_partitions, offstrategy, auto_trigger] (flat_mutation_reader reader) {
                     sstables::shared_sstable sst = use_view_update_path ? t->make_streaming_staging_sstable() : t->make_streaming_sstable_for_write();
                     schema_ptr s = reader.schema();
                     auto& pc = service::get_local_streaming_priority();
                     return sst->write_components(std::move(reader), adjusted_estimated_partitions, s,
                                                  t->get_sstables_manager().configure_writer("repair"),
                                                  encoding_stats{}, pc).then([sst] {
                         return sst->open_data();
-                    }).then([t, sst, offstrategy] {
+                    }).then([t, sst, offstrategy, auto_trigger] {
+                        if (auto_trigger) {
+                            rlogger.debug("Enabled automatic off-strategy trigger for table {}.{}",
+                                    t->schema()->ks_name(), t->schema()->cf_name());
+                            t->enable_off_strategy_trigger();
```

Comment:
with TWCS, where interporser creates multiple sstables, enable_off_strategy_trigger() will be called multiple times. I think we should do it only once.

Reply:
It is harmless to do it multiple times. It is also simpler not to track if it is the first time or second time. The second call will just update the timer.